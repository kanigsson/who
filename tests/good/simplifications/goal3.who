logic ->  : prop -> prop -> prop
logic = ['a||] : 'a -> 'a -> prop
type bool
logic true  : bool
logic false  : bool
goal ymemo_correct :
    forall (z : int).
      (forall x.  false = true -> 0 = x)
        -> 0 = 1
