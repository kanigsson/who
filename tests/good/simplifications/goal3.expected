logic ->  : prop -> prop -> prop
logic = ['a||] : a -> a -> prop
type bool
logic true  : bool
logic false  : bool
goal ymemo_correct : forall (z:int).
                     (forall (x:int).
                      (true =[bool||] false) -> (x =[int||] 0))
                     -> (0 =[int||] 1)
